src/Lorentz/Contracts/BaseDAO/Proposal.hs

```diff
@@ -604,63 +581,153 @@ unfreezeVoterToken = do
-    stackType @(Proposal pm : store : ProposalKey pm : [Operation] : s)
-
-    getField #pUpvotes
-    dip $ getField #pDownvotes
-    add; toNamed #proposalTotalVote
-
-    duupX @3
-    stToField #sQuorumThreshold; toNamed #sQuorumThreshold
-
-    if #sQuorumThreshold <=. #proposalTotalVote then do
+    dupTop2; doTotalVoteMeetQuorumThreshold
+    if Holds then do
       getFieldNamed #pUpvotes
       dip $ getFieldNamed #pDownvotes
       if #pUpvotes >. #pDownvotes then do
+        stackType @(Proposal pm : store : ProposalKey pm : [Operation] : s)
         push (#isAccepted .! True)
         callCachedFunc $ unfreezeProposerToken config
         unfreezeVoterToken
+
+        dup; dip swap
         cDecisionLambda
-        stackType @([Operation] : store : ProposalKey pm : [Operation] : s)
-        dig @3
-        iter cons
-        swap; dip swap
+
+        stackType @([Operation] : store : Proposal pm : ProposalKey pm : [Operation] : s)
+        dig @4
+        iter cons; swap
+        stackType @(store : [Operation] : Proposal pm : ProposalKey pm : s)
+        dig @3; dig @3
+        toField #pStartDate; deleteProposal
       else do
         -- Reject proposal if (upvote <= downvote)
+        stackType @(Proposal pm : store : ProposalKey pm : [Operation] : s)
         push (#isAccepted .! False)
         callCachedFunc $ unfreezeProposerToken config
         unfreezeVoterToken
-        drop
-
+        dip swap; toField #pStartDate; deleteProposal
     else do
-      -- Reject proposal regardless of upvotes
+      -- Reject proposal regardless of upvote
+      stackType @(Proposal pm : store : ProposalKey pm : [Operation] : s)
       push (#isAccepted .! False)
       callCachedFunc $ unfreezeProposerToken config
       unfreezeVoterToken
-      drop
+      dip swap; toField #pStartDate; deleteProposal
+  else do
+    -- Do nothing
+    drop; dip drop
 
-    dip $ drop @(ProposalKey pm)
+-- | Delete a proposal from 'sProposalKeyListSortByDate'
+-- Mark `pIsDeleted` to `True`
+deleteProposal
+  :: forall store ce pm s.
+    (NiceParameter pm, StorageC store ce pm)
+  => Timestamp : ProposalKey pm : store : s
+  :-> store : s
+deleteProposal = do
+  dip $ do dup; dip swap
+  -- Delete from 'sProposalKeyListSortByDate'
+  pair
+  dip $ stGetField #sProposalKeyListSortByDate
+  dip $ none
+  update
+  stSetField #sProposalKeyListSortByDate
 
-  else do
-    -- Add the proposalId back to #sProposalKeyListSortByDate
-    -- since we initially set this to []
-    dip drop
+  -- Mark as deleted
+  dup; duupX @3
+  stGet #sProposals
+  ifSome nop (failCustom_ #pROPOSAL_NOT_EXIST) -- should never happen
+  constructT @(Proposal pm)
+    ( fieldCtor $ getField #pUpvotes
+    , fieldCtor $ getField #pDownvotes
+    , fieldCtor $ getField #pStartDate
+    , fieldCtor $ getField #pMetadata
+    , fieldCtor $ getField #pProposer
+    , fieldCtor $ getField #pProposerFrozenToken
+    , fieldCtor $ getField #pVoters
+    , fieldCtor $ push True
+    )
```

Comment:
Can't we use just `setField` here? Invoking getters so many times is indeed expensive, maybe avoiding this would save us a lot of operation size.

Reply:
I did it that where since I actually don't know `setField` exist. I have updated this to use `setField` now. 

Tho, he operation size is reduced to 14880, the test still failed.